### PR TITLE
Replace chars when standard out can't be decoded

### LINKED ISSUE #2235: UnicodeDecodeError: 'utf-8' codec can't decode in run_command
Running FMRIPREP using singularity.

```
Node: _autorecon30
Working directory: /scratch/03763/oesteban/fmriprep-phase2/work/ds000172/sub-control01/fmriprep_wf/single_subject_control01_wf/anat_preproc_wf/surface_recon_wf/autorecon_resume_wf/autorecon3/mapflow/_autorecon
30

Node inputs:

T1_files = <undefined>
T2_file = <undefined>
args = <undefined>
big_ventricles = <undefined>
brainstem = <undefined>
directive = autorecon3
environ = {}
expert = <undefined>
flags = <undefined>
hemi = lh
hippocampal_subfields_T1 = <undefined>
hippocampal_subfields_T2 = <undefined>
hires = <undefined>
ignore_exception = False
mprage = <undefined>
mri_aparc2aseg = <undefined>
mri_ca_label = <undefined>
mri_ca_normalize = <undefined>
mri_ca_register = <undefined>
mri_edit_wm_with_aseg = <undefined>
mri_em_register = <undefined>
mri_fill = <undefined>
mri_mask = <undefined>
mri_normalize = <undefined>
mri_pretess = <undefined>
mri_remove_neck = <undefined>
mri_segment = <undefined>
mri_segstats = <undefined>
mri_tessellate = <undefined>
mri_watershed = <undefined>
mris_anatomical_stats = <undefined>
mris_ca_label = <undefined>
mris_fix_topology = <undefined>
mris_inflate = <undefined>
mris_make_surfaces = <undefined>
mris_register = <undefined>
mris_smooth = <undefined>
mris_sphere = <undefined>
mris_surf2vol = <undefined>
mrisp_paint = <undefined>
openmp = 8
parallel = <undefined>
subject_id = sub-control01
subjects_dir = /work/03843/crn_plab/stampede2/fmriprep-phase2/derivatives/ds000172/freesurfer
talairach = <undefined>
terminal_output = <undefined>
use_T2 = False
xopts = <undefined>

Traceback (most recent call last):
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/plugins/multiproc.py", line 51, in run_node
    result['result'] = node.run(updatehash=updatehash)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 407, in run
    self._run_interface()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 517, in _run_interface
    self._result = self._run_command(execute)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/pipeline/engine/nodes.py", line 650, in _run_command
    result = self._interface.run()
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1088, in run
    runtime = self._run_interface(runtime)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1654, in _run_interface
    runtime = run_command(runtime, output=self.terminal_output)
  File "/usr/local/miniconda/lib/python3.6/site-packages/niworkflows/nipype/interfaces/base.py", line 1423, in run_command
    result['stdout'] = stdout.decode(default_encoding).split('\n')
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc7 in position 9789: invalid continuation byte
```

### PR DESCRIPTION
**Pull request by oesteban**
Fixes #2235

### CODE REVIEW

**Comment by oesteban** [comment]
Tests are passing, should we go ahead?

**Comment by satra** [comment]
@oesteban - could we do a `try .. except`, so that we know when this happened, in case it stripped out meaningful things, and log that.

**Comment by oesteban** [comment]
@satra WDYT?

File: nipype/utils/filemanip.py [hunk 1]
```diff
@@ -596,6 +598,26 @@ def crash2txt(filename, record):
...
+
+    """
+    default_encoding = encoding or locale.getdefaultlocale()[1] or 'UTF-8'
+    logger = logger or fmlogger
+    try:
+        out = stream.decode(default_encoding)
+    except UnicodeDecodeError as err:
+        out = stream.decode(default_encoding, errors='replace')
+        logger.warning('Error decoding string: %s', err)
+    return out.split('\n')
```
Comment (effigies) [inline nipype/utils/filemanip.py]: `out.splitlines()` is robust to `'\n'` vs `'\r'` vs `'\r\n'`, if that's a possible issue.

**Comment by effigies** [comment]
This looks fine to me, but I'll leave the decision to @satra.

**Comment by satra** [comment]
LGTM as well - @oesteban - are you going to make the change @effigies suggested? i think this is ready for merge.

**Comment by oesteban** [comment]
Let's merge in 2h 30m when tests pass with @effigies suggestion :)

**Comment by effigies** [comment]
I think we can be pretty confident it will build successfully, so might as well build on master, rather than here and then on master. IMHO.

**Comment by oesteban** [comment]
Let's do it

**Pull request merged**